test/integ/axpy/src/axpy.cpp

```diff
@@ -239,14 +220,23 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(
     {
         auto const & val(pBufHostY[i]);
         auto const correctResult(alpha * pBufHostX[i] + pBufHostOrigY[i]);
-        if( boost::math::relative_difference(val, correctResult) > std::numeric_limits<Val>::epsilon() )
+        auto const relDiff = std::abs((val - correctResult) / std::min(val, correctResult));
+        if( relDiff > std::numeric_limits<Val>::epsilon() )
         {
             std::cerr << "C[" << i << "] == " << val << " != " << correctResult << std::endl;
             resultCorrect = false;
         }
     }
 
-    BOOST_REQUIRE_EQUAL(true, resultCorrect);
+    REQUIRE(resultCorrect);
 }
+};
 
-BOOST_AUTO_TEST_SUITE_END()
+TEST_CASE( "axpy", "[axpy]")
+{
+    using TestAccs = alpaka::test::acc::EnabledAccs<
+        alpaka::dim::DimInt<1u>,
+        std::size_t>;
+
+    alpaka::meta::forEachType< TestAccs >( TestTemplate() );
```

Comment:
This is a cool workaround but I will be missing the boost.test version where we had one test per type and could easily see if the test is failing for a specific type only.

Reply:
We could add a [section](https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#test-cases-and-sections) or some other kind of annotation.

Reply:
Update: Catch 2.5.0 added the `TEMPLATE_TEST_CASE` and `TEMPLATE_TEST_CASE_METHOD` macros: https://github.com/catchorg/Catch2/pull/1437

Reply:
Will this really help us? It looks like they support only expanded type lists. We are generating the type list dynamically so there still seems to be one step missing.

Reply:
Yes, it looks like it's close to what I build as well.